diff --git a/drivers/usb/core/quirks.c b/drivers/usb/core/quirks.c
index f681693f491e..934b3d997702 100644
--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -538,9 +538,6 @@ static const struct usb_device_id usb_quirk_list[] = {
 	/* INTEL VALUE SSD */
 	{ USB_DEVICE(0x8086, 0xf1a5), .driver_info = USB_QUIRK_RESET_RESUME },
 
-	/* Quectel EG25G Modem */
-	{ USB_DEVICE(0x2c7c, 0x0125), .driver_info = USB_QUIRK_RESET },
-
 	{ }  /* terminating entry must be last */
 };
 
