diff --git a/drivers/input/ff-memless.c b/drivers/input/ff-memless.c
index 8229a9006917..42edfad96456 100644
--- a/drivers/input/ff-memless.c
+++ b/drivers/input/ff-memless.c
@@ -26,10 +26,10 @@ MODULE_AUTHOR("Anssi Hannula <anssi.hannula@gmail.com>");
 MODULE_DESCRIPTION("Force feedback support for memoryless devices");
 
 /* Number of effects handled with memoryless devices */
-#define FF_MEMLESS_EFFECTS	16
+#define FF_MEMLESS_EFFECTS	64
 
 /* Envelope update interval in ms */
-#define FF_ENVELOPE_INTERVAL	50
+#define FF_ENVELOPE_INTERVAL	10
 
 #define FF_EFFECT_STARTED	0
 #define FF_EFFECT_PLAYING	1
